src/main/java/com/hermesworld/ais/galapagos/topics/controller/TopicController.java

```diff
@@ -96,6 +99,49 @@ public TopicController(ValidatingTopicService topicService, KafkaClusters kafkaE
         }
     }
 
+    @GetMapping(value = "/api/applicationsmetadata/{envId}/{topicName}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public List<KnownApplicationDto> getAvailableApplicationMetadata(@PathVariable String envId,
+            @PathVariable String topicName,
+            @RequestParam(name = "topicOwnerId", required = false) String topicOwnerId) {
+        TopicMetadata topic = topicService.getTopic(envId, topicName).orElseThrow(notFound);
+        List<String> producerIds = topic.getProducers();
+
+        return applicationsService.getAllApplicationMetadata(envId).stream().map(ApplicationMetadata::getApplicationId)
+                .filter(id -> !id.equals(topicOwnerId))
+                .filter(idsWithoutOwnerid -> !producerIds.contains(idsWithoutOwnerid))
+                .map(id -> toKnownAppDto(applicationsService.getKnownApplication(id)
+                        .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND))))
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping(value = "/api/producers/{environmentId}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public CompletableFuture<Void> addProducerToTopic(@PathVariable String environmentId,
+            @RequestBody AddProducerDto producers) {
+        if (producers.getProducers().stream().anyMatch(appId -> !applicationsService.isUserAuthorizedFor(appId))) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+        }
+
+        if (producers.getProducers().isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        return topicService.addTopicProducers(environmentId, producers.getTopicName(), producers.getProducers());
+    }
+
+    @DeleteMapping(value = "/api/producers/{envId}/{topicName}/{producerApplicationId}")
+    public CompletableFuture<Void> removeProducerFromTopic(@PathVariable String envId, @PathVariable String topicName,
+            @PathVariable String producerApplicationId) {
+        if (envId.isEmpty() || topicName.isEmpty() || producerApplicationId.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        if (!applicationsService.isUserAuthorizedFor(producerApplicationId)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+        }
+
+        return topicService.removeProducerFromTopic(envId, topicName, producerApplicationId);
```

Comment:
This method should return an HTTP 204 (NO_CONTENT) status. This is common for REST endpoints deleting something.